src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java

```diff
@@ -105,76 +126,105 @@ public Future<ConnectionStatus> getConnectionStatus() {
+                .state(ConnectedState.FAILED)
+                .errors(
+                    new AuthErrors().withSignIn(
+                        "Failed to connect to Kafka cluster: %s".formatted(cause.getMessage())
+                    )
+                ).build()
+        );
+      }
+    }).orElseGet(() -> {
+      // There is no Kafka cluster configuration, so return the NONE state
       return Future.succeededFuture(
           ConnectionStatusKafkaClusterStatusBuilder
               .builder()
               .state(ConnectedState.NONE)
               .build()
       );
-    }
-    // There is a Kafka configuration, so validate the connection by creating an AdminClient
-    // and describing the cluster.
-    try (var adminClient = createAdminClient(kafkaConfig)) {
-      var clusterDesc = adminClient.describeCluster();
-      var actualClusterId = clusterDesc.clusterId().get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
-      // Set the cluster ID
-      getSpec().kafkaClusterConfig().withId(actualClusterId);
-      return Future.succeededFuture(
-          ConnectionStatusKafkaClusterStatusBuilder
-              .builder()
-              .state(ConnectedState.SUCCESS)
-              .build()
-      );
-    } catch (Exception e) {
-      // The connection failed, so successfully return the failed state
-      return Future.succeededFuture(
-          ConnectionStatusKafkaClusterStatusBuilder
-              .builder()
-              .state(ConnectedState.FAILED)
-              .errors(
-                  new AuthErrors().withSignIn(
-                      "Failed to connect to Kafka cluster: %s".formatted(e.getMessage())
-                  )
-              ).build()
-      );
-    }
+    });
   }
 
   protected Future<SchemaRegistryStatus> getSchemaRegistryConnectionStatus() {
-
-    var schemaRegistryConfig = spec.schemaRegistryConfig();
-    if (schemaRegistryConfig == null) {
+    return withSchemaRegistryClient(srClient -> {
+      // There is a configuration, so validate the connection by creating a SchemaRegistryClient
+      // and getting the global mode.
+      try {
+        srClient.getMode();
+        return Future.succeededFuture(
+            ConnectionStatusSchemaRegistryStatusBuilder
+                .builder()
+                .state(ConnectedState.SUCCESS)
+                .build()
+        );
+      } catch (Exception e) {
+        // The connection failed, so successfully return the failed state
+        return Future.succeededFuture(
+            ConnectionStatusSchemaRegistryStatusBuilder
+                .builder()
+                .state(ConnectedState.FAILED)
+                .errors(
+                    new AuthErrors().withSignIn(
+                        "Failed to connect to Schema Registry: %s".formatted(e.getMessage())
+                    )
+                ).build()
+        );
+      }
+    }).orElseGet(() -> {
+      // There is no Schema Registry configuration, so return the NONE state
```

Comment:
This just refactors the `DirectConnectionState` to have a public `withSchemaRegistryClient(...)` method and to use that here. The `RealDirectFetcher` can then use the `withSchemaRegistryClient(...)`, avoiding some duplicate code.